frontend/pages/profile/index.js

```diff
@@ -0,0 +1,62 @@
+import { useRouter } from 'next/router';
+import dynamic from 'next/dynamic';
+import resume from '../../resume.json'; // Import your resume JSON data
+
+const Profile = ({ resumeData }) => {
+  const router = useRouter();
+  const { template } = router.query;
+  const { basics, work, projects, skills, awards } = resumeData; // Destructure data from props
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rohankokkulabito/resfolio/pull/3#issuecomment-2230682191>#edaf96</a><span> - 07/16/2024, 12:05 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'useRouter' hook is called conditionally based on the presence of the 'template' query parameter. This violates the rules of hooks in React, which state that hooks should not be called conditionally.
</div>
<div id="fix">
<b>Fix:</b> Ensure that the 'useRouter' hook is called unconditionally at the top level of the component.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -5,12 +5,12 @@
  const Profile = ({ resumeData }) => {
    const router = useRouter();
 -  const { template } = router.query;
    const { basics, work, projects, skills, awards } = resumeData; // Destructure data from props
   
 +  const { template } = router.query;
    if (!template) {
      return <div>Loading...</div>;
    }
   
    const templateName = template.charAt(0).toUpperCase() + template.slice(1);
   
    // Dynamically import template components based on the selected template
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged